### PR TITLE
LPS-112983 Deprecate liferay-browser-selectors

### PR DESCRIPTION
**Pull request by jbalsas**
Hey @wincent, @julien, this is a revamp of https://github.com/wincent/liferay-portal/pull/291

**Done:**
- Inlined all references to `.touch` and `.touchMobile` replacing them with `.touchEnabled` or similars so we don't have a runtime implicit dependency on `liferay-browser-selectors`.
- Made `liferay-browser-selectors` optional
- Added a `frontend-theme-browser-support` module (because naming is hard and overmodularizing is the way!). The module implements [this suggestion](https://github.com/wincent/liferay-portal/pull/291#issuecomment-633417869) and adds the `{chrome|ie|edge|firefox|other}` and ` mobile` classes to the `body` element. This is a change from the old behaviour that was putting them in the `html` tag. I could only see `html.rtl` being used, so I figure this is fair game, but this could potentially break stuff... we can add support for `root_css_class` to be extended as well if we decide it's worth avoiding the breaking change (but that will introduce a breaking change of a different nature).

**Pending:**
- Deprecate the module (assuming we're doing so)
- The `.touch` class is not added. Usage is in 3 css spots. I'd be inclined to just look for smoke here and then figure out a better way to handle this, but I'm open to suggestions :)
- Check if additional classes need to be added or removed elsewhere

Feel free to take it from here! ;)

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/layout.js [hunk 1]
```diff
@@ -158,8 +158,7 @@ AUI.add(
 
 					if (!firstPortletStatic || firstPortletStatic == 'end') {
 						referencePortlet = firstPortlet;
-					}
-					else {
+					} else {
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/layout.js:161]: Going to need an SF of this — same in some other place(s) as well.

File: modules/apps/frontend-theme/frontend-theme-browser-support/src/main/java/com/liferay/frontend/theme/browser/support/internal/theme/contributor/BrowserTemplateContextContributor.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	@Override
+	public void prepare(
+		Map<String, Object> contextObjects,
+		HttpServletRequest httpServletRequest) {
+
+		StringBuilder sb = new StringBuilder();
+
+		sb.append(GetterUtil.getString(contextObjects.get("bodyCssClass")));
+		sb.append(StringPool.SPACE);
+		sb.append(BrowserSnifferUtil.getBrowserId(httpServletRequest));
```
Comment (wincent) [inline modules/apps/frontend-theme/frontend-theme-browser-support/src/main/java/com/liferay/frontend/theme/browser/support/internal/theme/contributor/BrowserTemplateContextContributor.java:48]: Only weird thing about this is if it isn't one of the browsers detected by `getBrowserId()` we're going to get an `.other` class on the `body`. No harm done, I guess, but I'd be surprised if we ever targeted such a class.

**Comment by wincent** [comment]
I took a look at this @julien. Looks ok to me (other than the SF thing). Do you want to take care of the remaining steps after you wrap up your current tasks?

**Comment by julien** [comment]
@wincent I'm currently doing an `ant all` to test this out, and yes I'll take care of the remaining steps for this one. Thanks

File: modules/apps/frontend-theme/frontend-theme-browser-support/src/main/java/com/liferay/frontend/theme/browser/support/internal/theme/contributor/BrowserTemplateContextContributor.java (hunk 2 above)
Comment (jbalsas) [reply modules/apps/frontend-theme/frontend-theme-browser-support/src/main/java/com/liferay/frontend/theme/browser/support/internal/theme/contributor/BrowserTemplateContextContributor.java:48]: Yup!

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/layout.js (hunk 1 above)
Comment (jbalsas) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/layout.js:161]: Might be my VSCode Prettier setup is not correct 😭

Comment (wincent) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/layout.js:161]: toughen-up

**Comment by julien** [comment]
@wincent, @jbalsas I sent a follow up [here](https://github.com/wincent/liferay-portal/pull/296)

**Comment by julien** [comment]
ci:close

**Pull request closed without merging**